spec/lib/personas/tools/create_image_spec.rb

```diff
@@ -0,0 +1,118 @@
+# frozen_string_literal: true
+
+RSpec.describe DiscourseAi::Personas::Tools::CreateImage do
+  let(:prompts) { ["a watercolor painting", "an abstract design"] }
+
+  fab!(:gpt_35_turbo) { Fabricate(:llm_model, name: "gpt-3.5-turbo") }
+
+  before do
+    SiteSetting.ai_bot_enabled = true
+    toggle_enabled_bots(bots: [gpt_35_turbo])
+    SiteSetting.ai_openai_api_key = "abc"
+  end
+
+  let(:bot_user) { DiscourseAi::AiBot::EntryPoint.find_user_from_model(gpt_35_turbo.name) }
+  let(:llm) { DiscourseAi::Completions::Llm.proxy("custom:#{gpt_35_turbo.id}") }
+  let(:progress_blk) { Proc.new {} }
+
+  let(:create_image) { described_class.new({ prompts: prompts }, llm: llm, bot_user: bot_user) }
+
+  let(:base64_image) do
+    "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg=="
+  end
+
+  describe "#process" do
+    it "can generate images with gpt-image-1 model" do
+      data = [{ b64_json: base64_image, revised_prompt: "a watercolor painting of flowers" }]
+
+      WebMock
+        .stub_request(:post, "https://api.openai.com/v1/images/generations")
+        .with do |request|
+          json = JSON.parse(request.body, symbolize_names: true)
+
+          expect(prompts).to include(json[:prompt])
+          expect(json[:model]).to eq("gpt-image-1")
+          expect(json[:size]).to eq("auto")
+          true
+        end
+        .to_return(status: 200, body: { data: data }.to_json)
+
+      info = create_image.invoke(&progress_blk).to_json
+
+      expect(JSON.parse(info)).to eq(
+        {
+          "prompts" => [
+            {
+              "prompt" => "a watercolor painting of flowers",
+              "url" => "upload://pv9zsrM93Jz3U8xELTJCPYU2DD0.png",
+            },
+            {
+              "prompt" => "a watercolor painting of flowers",
+              "url" => "upload://pv9zsrM93Jz3U8xELTJCPYU2DD0.png",
```

Comment:
Sorry as this is confusing to me. Why do we return two prompts with the same `prompt` and `url` attributes?

Reply:
it got remapped by the model